### DISCUSSION TITLE
No notification sound in IOS

### DISCUSSION THREAD

**Discussion by kristijorgji** [kreait/firebase-php#634] [Q&A]
Using `"kreait/firebase-php": "^4.18",`

In usage am specifyind appnsconfig.payload.aps.sound = 'default' but no sound is used in ios on receiving notification

Code below
```
$message = CloudMessage::withTarget('token', $subscription->getToken())
                ->withNotification(Notification::create(
                    null,
                    $data['message']
                ))
                ->withData($data)
                ->withAndroidConfig([
                        'notification' => [
                            'sound' => 'default'
                        ]
                    ]
                )->withApnsConfig(ApnsConfig::fromArray([
                    'payload' => [
                        'aps' => [
                            'badge' => 1,
                            'sound' => 'default',
                        ],
                    ],
                ]));
```

**Comment by jeromegamez** [comment]
Issues like this one are hard to debug from an Admin SDK maintainer's view 😅/😓 - there are just too many reasons why no sounds are played - perhaps it's because the notification has no title, it could also be one or a combination of different notification setting(s) on the iOS client (DnD, Focus mode, silent mode, app settings). While I would like to be able to help, there's not much I can do here to help - all I can say is that if the Firebase FCM API accepts the message, it's valid - the delivery by Firebase (the service) and handling on the client is out of my reach :/

**Reply by kristijorgji** [reply]
Hi, thanks for reply. Sure I understand that, just wanted  confirmatiion that for my version of library `"kreait/firebase-php": "^4.18",

` code was correct. I saw that code snippet for setting apns config in documentation but was for last version not one i am using that is mostly why i asked. Maybe that iss reason I get no sound.. Other then that the ios device is my personal one for testing, has enabled notification and no restriction. Notiification comes in just fine, only sound is not played. 

So question then would be, is  at least that code correct ? and does the `default` value for sound is valid

**Reply by jeromegamez** [reply]
The code looks fine to me (the notification title set to `null` threw me off in the beginning) - the SDK only checks for the most common misconfigurations, and especially for the Android and APNs payloads does very little checks. Perhaps you could try to raise the apns-priority level in the apns header to see if that helps. Besides that I don't have much of an idea - but

* APNS Payload reference: https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html#//apple_ref/doc/uid/TP40008194-CH17-SW1
* APNS Header reference: https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html#//apple_ref/doc/uid/TP40008194-CH11-SW13

If you're not sure if something is supported in the latest 4.x version in reference to the 5.x docs, you can see the latest 4.x documentation at https://firebase-php.readthedocs.io/en/4.x/ (there's a version switcher in the bottom left of the pages).
